src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -249,7 +249,12 @@ private void SetUrl()
                 }
 
                 EventSourceTrace("Url: {0}", requestUri);
-                SetCurlOption(CURLoption.CURLOPT_URL, requestUri.AbsoluteUri);
+                string idnHost = requestUri.IdnHost;
```

Comment:
Any idea how much expense this adds?  I took a quick look at IdnHost, and there's a lot of code in there, but I didn't look to see how much overhead it would actually add and whether it will be impactful or not.